src/controls/Button.tsx

```diff
@@ -59,35 +55,35 @@ class ClickActionProperty extends ControlProperty<string> {
     return {
       elementToMount: fragment as any,
     };
-  }
-}
-
-export class Button extends Control {
-  private buttonElement: HTMLButtonElement;
-
-  @controlProperty(new TextAlignmentProperty((c: Button) => c.buttonElement))
-  public text: string;
-
-  @controlProperty(new TextFormattingProperty((c: Button) => c.buttonElement))
-  public textFormatting: Formatting;
+  },
+};
 
-  @controlProperty(new FontSizeProperty((c: Button) => c.buttonElement))
-  public fontSize: number;
-
-  @controlProperty(new TextContentProperty((c: Button) => c.buttonElement))
-  public textAlignment: string;
-
-  @controlProperty(new ClickActionProperty((c: Button) => c.buttonElement))
-  public clickScript: string;
-
-  protected initialize(): HTMLElement {
-    this.buttonElement = document.createElement('button');
-    return this.buttonElement;
-  }
+interface IButton extends IControlWithText {
+  clickScript: string;
+}
 
-  public get descriptor(): IControlDescriptor<Control> {
-    return buttonDescriptor;
-  }
+interface IButtonState {
+  root: HTMLElement;
 }
 
-export let buttonDescriptor = new ReflectionBasedDescriptor('button', Button);
+export const Button = createControlDefinition<IButton, IButtonState>({
+  id: 'button',
+  displayName: 'Button',
+  factory: function () {
+    let button = document.createElement('button');
+    return {
+      root: button,
+      button: button,
+    };
+  },
+}).defineProperties((meta) => ({
+  text: meta.compose(TextContentProperty, 'root'),
```

Comment:
Composition is a good idea - notice that TextContent is targeting an HtmlElement, but the "state" here is a bag of properties.  Thus the `meta.compose` took the state of the Button, took the `root` element, then passed the resulting value (the HtmlElement) to `TextContentProperty`.

Compared to our current `ControlProperty` this seems better - it avoids the odd `setValueRaw` etc. which was added to make implementations easier - they're only concerned with an HtmlElement instead of the control itself